src/sidebar/components/HypothesisApp.js

```diff
@@ -162,24 +179,73 @@ function HypothesisApp({ auth, frameSync, settings, session, toastMessenger }) {
     session.logout();
   };
 
+  /**
+   * Style a div as a scroll-root container for sidebar content.
+   *
+   * @param {PresentationalDivProps} props
+   */
+  function SidebarScrollContainer({
+    children,
+    classes,
+    elementRef,
+    ...restProps
+  }) {
+    return (
+      <div
+        className={classnames(
+          'h-full min-h-full overflow-scroll bg-grey-2',
+          // Precise padding alignment with sidebar content; extra padding
+          // at the bottom on wide screens
+          'p-[9px] lg:pb-16',
+          'js-thread-list-scroll-root',
+          classes
+        )}
+        {...restProps}
+        ref={downcastRef(elementRef)}
+      >
+        {children}
+      </div>
+    );
+  }
+
+  /**
+   * Open the help panel, or, if a service callback is configured to handle
+   * help requests, fire a relevant event instead
+   */
+  const onRequestHelp = useCallback(() => {
+    const service = serviceConfig(settings);
+    if (service && service.onHelpRequestProvided) {
+      frameSync.notifyHost('helpRequested');
+    } else {
+      store.toggleSidebarPanel('help');
+    }
+  }, [frameSync, settings, store]);
+
   return (
-    <div
-      className={classnames('HypothesisApp', 'js-thread-list-scroll-root', {
+    <SidebarScrollContainer
+      classes={classnames({
```

Comment:
Extracting `SidebarScrollContainer` but also passing some extra classes here helps me delineate between the underlying design pattern and the customization of it based on route or theming. It's a helpful mental model.